pkgs/os-specific/linux/systemd/0021-add-rootprefix-to-lookup-dir-paths.patch

```diff
@@ -0,0 +1,38 @@
+From 8d1618a97ad08078815f409f03b45aff3ae6bd0a Mon Sep 17 00:00:00 2001
+From: Andreas Rammhold <andreas@rammhold.de>
+Date: Thu, 9 May 2019 11:15:22 +0200
+Subject: [PATCH 21/27] add rootprefix to lookup dir paths
+
+systemd does not longer use the UDEVLIBEXEC directory as root for
+discovery default udev rules. By adding `$out/lib` to the lookup paths
```

Comment:
Im still not sure if this is desired behaviour? don't we want to only read `udev` rules from `/etc/`.  e.g. do something similar like we do for `unit` files where we explicitly say which upstream files from the package get `symlinked` into `/etc/udev` ? This is something that has confused me before.   Not something for this PR; but we could open an issue on this

Reply:
This is to have systemd/udev discover the udev rules that come with it:
```
50-udev-default.rules
60-autosuspend-chromiumos.rules
60-block.rules
60-cdrom_id.rules
60-drm.rules
60-evdev.rules
60-fido-id.rules
60-input-id.rules
60-persistent-alsa.rules
60-persistent-input.rules
60-persistent-storage.rules
60-persistent-storage-tape.rules
60-persistent-v4l.rules
60-sensor.rules
60-serial.rules
61-autosuspend-manual.rules
64-btrfs.rules
70-joystick.rules
70-mouse.rules
70-power-switch.rules
70-touchpad.rules
70-uaccess.rules
71-seat.rules
73-seat-late.rules
75-net-description.rules
75-probe_mtd.rules
78-sound-card.rules
80-drivers.rules
80-net-setup-link.rules
90-vconsole.rules
99-systemd.rules
```

These are different from the ones installed via `udev.packages` (which end up in `/etc/udev/rules.d`)

I personally like the distinction beween systemd-provided and "system config provided" - by symlinking, we'd loose this. So far, there hasn't been the need to make this configurable, and it's not required for the bumps here either. We should probably discuss this in https://github.com/NixOS/nixpkgs/issues/80038, as well as the relevance of the other patches.